src/compiler/commandLineParser.ts

```diff
@@ -441,6 +449,7 @@ namespace ts {
                 type: "string",
                 isFilePath: true
             },
+            affectsSourceFile: true,

```

Comment:
Why ? RootDir and paths shouldn't affect syntaxTree or bindDiagnostics?
Same goes for lib, baseUrl, typeRoots, paths

Reply:
Those all effect module resolution, and `SourceFile` stores `resolvedModules`.